src/quickdraw.c

```diff
@@ -101,11 +101,11 @@ draw_oscil_plot(mrb_state *mrb, mrb_value self)
 
     int off = phase * (n-1);
 
-
     const int min_y = bound_y;
     const int max_y = bound_y + bound_h;
 
-    float ypos = bound_x + bound_h/2 - bound_h/2*f[off];
+    int ii = off%n;
+    float ypos = -bound_h/2*f[ii] + bound_y+bound_h/2.0;
```

Comment:
Could ii possibly end up being negative here!?

Reply:
n should be strictly greater than 0. From what I can tell the phase variable should be 0..N, though there aren't any asserts backing that claim up.